include/boost/unordered/detail/foa.hpp

```diff
@@ -1217,13 +1217,22 @@ table:empty_value<Hash,0>,empty_value<Pred,1>,empty_value<Allocator,2>
 
   table& operator=(const table& x)
   {
+    BOOST_STATIC_ASSERT(boost::is_nothrow_swappable<Hash>::value);
+    BOOST_STATIC_ASSERT(boost::is_nothrow_swappable<Pred>::value);
+
     static constexpr auto pocca=
       alloc_traits::propagate_on_container_copy_assignment::value;
 
     if(this!=std::addressof(x)){
+      hasher    tmp_h=x.h();
+      key_equal tmp_p=x.pred();
+
+      using std::swap;
+      swap(h(),tmp_h);
+      swap(pred(),tmp_p);
```

Comment:
Explanatory comments are missing here. "// can throw" on the tmp lines, then "// swap operations don't throw, so we know we have consistent hasher and key_equal at this point." Or even more verbose.

Reply:
Hopefully these comments offer a bit more clarity.